src_dualdesc/POLY_MPI_DualDesc.cpp

```diff
@@ -26,7 +26,10 @@ int main(int argc, char *argv[]) {
   // outside of the lifetime of world creates a MPI_ABORT.
   // ---We should also avoid using exit when terminating
   // because it makes an unscheduled destruction of the communicator.
-  boost::mpi::environment env;
+  boost::mpi::environment env(boost::mpi::threading::serialized);
```

Comment:
Can you document a little bit what is this boost::mpi::threading::serialized and what it implies to
the system? I can't find by a google search what that mean.

Reply:
I had to dig into the boost code to find out about the difference, but the documentation there is quite clear.
Essentially we need this because the communication (sub-)thread makes MPI calls, but never at the same time as the main thread.
```
enum level {
  /** Only one thread will execute. 
   */
  single     = MPI_THREAD_SINGLE,
  /** Only main thread will do MPI calls.
   * 
   * The process may be multi-threaded, but only the main 
   * thread will make MPI calls (all MPI calls are ``funneled''
   * to the main thread).
   */
  funneled   = MPI_THREAD_FUNNELED,
  /** Only one thread at the time do MPI calls.
   * 
   * The process may be multi-threaded, and multiple 
   * threads may make MPI calls, but only one at a time:
   * MPI calls are not made concurrently from two distinct 
   * threads (all MPI calls are ``serialized'').
   */
  serialized = MPI_THREAD_SERIALIZED,
  /** Multiple thread may do MPI calls.
   * 
   * Multiple threads may call MPI, with no restrictions.
   */
  multiple   = MPI_THREAD_MULTIPLE
};
```

Reply:
Ok, thank you.
Please add something like that to the code.

There is always a balance when putting comments, but if something takes 10 minutes to find
explanation or to understand, then I think comments are helpful.